library/manageiq_provider.py

```diff
@@ -531,16 +551,18 @@ def main():
             provider_region=dict(required=False, type='str'),
             access_key_id=dict(required=False, type='str', no_log=True),
             secret_access_key=dict(required=False, type='str', no_log=True),
-            metrics=dict(required=False, type='bool', default=False),
-            hawkular_hostname=dict(required=False),
-            hawkular_port=dict(required=False),
+            monitoring=dict(required=False, default=None,
+                       choices=['hawkular', 'prometheus', None]),
+            monitoring_hostname=dict(required=False),
+            monitoring_port=dict(required=False),
             initiate_refresh=dict(required=False, type='bool', default=True),
             validate_provider_auth=dict(required=False, type='bool', default=True)
         ),
         required_if=[
             ('provider_type', 'openshift-origin', ['provider_api_hostname', 'provider_api_port', 'provider_api_auth_token']),
             ('provider_type', 'openshift-enterprise', ['provider_api_hostname', 'provider_api_port', 'provider_api_auth_token']),
-            ('metrics', True, ['hawkular_hostname', 'hawkular_port']),
+            ('monitoring', 'hawkular', ['monitoring_hostname', 'monitoring_port']),
+            ('monitoring', 'prometheus', ['monitoring_hostname', 'monitoring_port']),
```

Comment:
I don't think there is.
https://github.com/ansible/ansible/blob/b69c7f50d02dac3e24bf1927ebdc933475da93b7/lib/ansible/module_utils/basic.py#L1635 checks `== val`.
Are you unhappy with textual duplication (LGTM) or the fact we'll have to remember to extend it if we add another?

required_if is not magical.
If you want you can check the condition yourself and exit with error (like done below for  `['miq_url', 'miq_username', 'miq_password']`)...